BibBuddy/app/src/main/java/de/bibbuddy/BookOnlineFragment.java

```diff
@@ -91,25 +92,32 @@ private void closeFragmentAfterAdding() {
   }
 
   private void handleIsbnInput(int keyCode, KeyEvent event) {
+    String textInput = searchFieldText.getText().toString().replaceAll("\\s", "");
+
     if (keyCode == KeyEvent.KEYCODE_ENTER) {
       if (event.getAction() != KeyEvent.ACTION_DOWN) {
-        isbnRetriever = new IsbnRetriever(searchFieldText.getText().toString());
-        thread = new Thread(isbnRetriever);
-        thread.start();
-
-        try {
-          thread.join();
-        } catch (Exception e) {
-          System.out.println(e);
-        }
-
-        // retrieve metadata that was saved
-        Book book = isbnRetriever.getBook();
-        List<Author> authors = isbnRetriever.getAuthors();
-        if (book != null) {
-          handleAddBook(book, authors);
+        if (dataValidation.isValidIsbn10or13(textInput)) {
+          isbnRetriever = new IsbnRetriever(searchFieldText.getText().toString());
+          thread = new Thread(isbnRetriever);
+          thread.start();
+
+          try {
+            thread.join();
+          } catch (Exception e) {
+            System.out.println(e);
```

Comment:
```suggestion
            e.printStackTrace();
```
Log exceptions by printStackTrace instead of custom system prints.

Reply:
Done.